std/logger/filelogger.d

```diff
@@ -80,7 +80,10 @@ class FileLogger : Logger
                                    " created in '", d,"' could not be created."));
         }
 
-        this.file_.open(this.filename, "a");
+        static if (is(This == shared))
+            () @trusted { (cast() this.file_).open(this.filename, "a"); }();
```

Comment:
Just do this all the time? Then no need to template the constructor, add a comment about why.

Reply:
The constructor still needs to be templated, though.